### PR TITLE
Almost complete Airport Challenge

### PR DESCRIPTION
**Pull request by amarzook1**

### CODE REVIEW

**Comment by EdwardAndress** [comment]
Hey Ahmed!  Nice effort.  I'm going to leave some comments inline against the code.  Watch this space 👀 !

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require_relative 'plane.rb'
+require_relative 'weather.rb'
+
+class Airport
+  
+  def initialize
+    @planes = []
+    @parked = false
+    @weather = Weather.new
```
Comment (EdwardAndress) [inline lib/airport.rb]: Good call to make a weather class.  The next step would be to use dependency injection.  See if you can find out what that means and how to use it.

File: lib/airport.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require_relative 'plane.rb'
+require_relative 'weather.rb'
+
+class Airport
+  
+  def initialize
+    @planes = []
+    @parked = false
```
Comment (EdwardAndress) [inline lib/airport.rb]: What does `@parked` refer to?  Whether or not there are planes in the hangar?

File: lib/airport.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  def takeoff
+    fail 'Weather is Stormy' if stormy?
+    @parked = false
+    @planes.pop
+  end
+  
+  attr_reader :parked
+  
+  def stormy?
+    @weather == true
```
Comment (EdwardAndress) [inline lib/airport.rb]: `@weather` refers to an instance of the `Weather` class.  So `@weather == true` is never going to return true.  How can you fix this by using one of the methods defined in the `Weather` class?

File: spec/weather_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require "weather.rb"
+
```
Comment (EdwardAndress) [inline spec/weather_spec.rb]: We covered some stuff about stubbing in the code review, which should help you to improve your tests here.

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
+require 'airport'
+
+describe Airport do
+  
+  it 'responds to land' do
+    expect(subject).to  respond_to(:land).with(1).argument
+  end
+
```
Comment (EdwardAndress) [inline spec/airport_spec.rb]: The typical RSpec structure is to have a `describe` block for each method you're testing, then an `it` block for each test relating to that method.

Something along these lines...

```
describe '#land' do
  it 'lands a plane and adds it to the planes array' do
    subject.land(plane)
    expect(subject.planes).to eq [plane]
  end
end
```

File: spec/plane_spec.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+describe Plane do
```
Comment (EdwardAndress) [inline spec/plane_spec.rb]: If there's no behaviour to encapsulate, maybe you don't yet need a plane class.

File: lib/airport.rb (hunk 3 above)
Comment (amarzook1) [reply lib/airport.rb]: @weather.check

File: lib/airport.rb (hunk 2 above)
Comment (amarzook1) [reply lib/airport.rb]: yeah thinking of renaming it to location for better readability and putting it in the plane class instead to refactor better

File: spec/plane_spec.rb (hunk 6 above)
Comment (amarzook1) [reply spec/plane_spec.rb]: going to add the location to the plane class instead to refactor my code a bit.

**Pull request closed without merging**